trecs/models/mf.py

```diff
@@ -0,0 +1,293 @@
+"""
+Implicit MF recommender system
+"""
+import numpy as np
+from lenskit.algorithms import als
+import pandas as pd
+from trecs.metrics import MSEMeasurement
+from trecs.random import Generator
+from trecs.validate import validate_user_item_inputs
+from trecs.utils import non_none_values
+from .recommender import BaseRecommender
+
+
+class ImplicitMF(BaseRecommender):
```

Comment:
If we assume that we might want to integrate other MF models later on (e.g., one for explicit data, or one of the other LensKit models), would we want that to happen through optional arguments to a general MF class or have each be its own MF class?

Reply:
I think we'd want each to be its own class. What might be worthwhile is having an `MF` abstract class that extends `BaseRecommender`, and then have `ImplicitMF` implement that abstract class. Future classes can also implement the `MF` abstract class. I'm down to punt on this to the future when someone makes a feature for handling explicit ratings, since that will likely be it own large PR...

Reply:
That makes sense.